app/uk/gov/hmrc/exports/repositories/UpdateSubmissionsTransactionalOps.scala

```diff
@@ -167,24 +137,108 @@ class UpdateSubmissionsTransactionalOps @Inject() (
     submissionRepository.findOneAndUpdate(session, BsonDocument(filter.toString), BsonDocument(update.toString))
   }
 
-  private def updateSubmissionRequest(
+  private def updateAmendmentRequest(
     session: ClientSession,
-    actionId: String,
+    action: Action,
+    mrn: String,
+    summary: NotificationSummary,
+    actions: Seq[Action],
+    submissionStatus: SubmissionStatus,
+    latestEnhancedStatus: EnhancedStatus
+  ): Future[Option[Submission]] = (for {
+    decId <- action.decId
+    update <- submissionStatus match {
```

Comment:
The two cases of this `match` do the same thing apart the different `latestEnhancedStatus`. Why not deriving it before calling `updateAmendmentRequest` in order to have a single `Json.obj(...` ?
In case `SubmissionStatus` is not `REJECTED`, `CUSTOMS_POSITION_DENIED` or `CUSTOMS_POSITION_GRANTED` you could even avoid to call the method. Just a suggestion btw.